### DISCUSSION TITLE
Does Zeit/now support Pull request previews?

### DISCUSSION THREAD

**Discussion by Andre601** [vercel/vercel#3992] [General] [answered]
If I understand this repository correctly is zeit/now used to build changes to then deploy on a domain (Similar to what f.e. netlify does)?
If that's the case, can it be used for previewing changes of Pull requests? I want to allow people to PR changes to some documentation I have, but I don't want to pull those changes and build it manually and rather have a tool doing this for me... Netlify is a bit meh on describing this and I found this one here by shear coincidence and was now wondering, if I can use it for the above mentioned purpose.

**Comment by styfle** [comment, marked as answer]
Yes of course!

Take a look here: https://vercel.com/docs/git/vercel-for-github

**Reply by Andre601** [reply]
I face some minor issues, as it seems to not be usable for organisations?

**Reply by styfle** [reply]
Yes its usable for organizations. You need to create a ZEIT Team to connect your GitHub Organization.

**Reply by Andre601** [reply]
Alright, I now have this.
Another question: I use MkDocs for building static sites. The "Build Command" is obvious, as it would be `mkdocs build` in my case, but I'm not quite sure what "Development Command" means. For what situations would this be actually used?

Additionally do I probably need to define some dependencies for the deployment?

**Reply by styfle** [reply]
You only need a "Development Command" if you plan to add [Serverless Functions](https://zeit.co/docs/v2/serverless-functions/introduction). This is the command that would run with during `now dev` locally.

Looking at the [mkdocs documentation](https://github.com/mkdocs/mkdocs/blob/2505a907f4ef6cc4492031faef06192f9eddf0bc/docs/user-guide/configuration.md#dev_addr), the dev command would probably be:

```
mkdocs serve --dev-addr $PORT
```

**Comment by Andre601** [comment]
Do I require to create a specific file in the repository in order to make Zeit now work with the "Other" option, as mkdocs isn't a supported option?

**Reply by styfle** [reply]
No, just the `Build Command` which is likely `mkdocs build`.

I'll see if we can add auto-detection for this framework by reading the `requirements.txt` file.

**Reply by Andre601** [reply]
I did this. but received errors, where it didn't recognize the command.

Output from the logs:
```
23:12:17.367  sh: mkdocs: command not found
23:12:17.370  Error: Exited with 127
23:12:17.370      at ChildProcess.<anonymous> (/zeit/f73854a5c95ea7a1/.build-utils/node_modules/@now/build-utils/dist/index.js:31349:24)
23:12:17.370      at ChildProcess.emit (events.js:223:5)
23:12:17.370      at ChildProcess.EventEmitter.emit (domain.js:475:20)
23:12:17.370      at maybeClose (internal/child_process.js:1021:16)
23:12:17.370      at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

**Reply by styfle** [reply]
Our system doesn't have mkdocs installed.

If you add a `requirements.txt` file with `mkdocs==1.0.4` then it will install the specified version.

https://github.com/zeit/now/blob/master/packages/now-static-build/test/fixtures/41-mkdocs/requirements.txt

**Reply by Andre601** [reply]
Thank you! It now worked.
Last question (I hope): Would it be possible to set this up, to also automatically build and push changes to GitHub-pages (the `gh-pages` branch) when the PR is merged?
Or would it be better to make a GitHub action for this stuff?

**Reply by styfle** [reply]
You can link specific branches to specific domains so you could use `gh-pages` instead of `master` as your production domain if you wish.

https://zeit.co/blog/branch-domains
